### ISSUE TITLE
KLVP on Linux

### ISSUE THREAD

**Issue by Harrold** [jimcavoy/klvp#23]
I get following error message when I run 
cmake -S . -B ./build
on Linux.

CMake Error at klvp/CMakeLists.txt:27 (find_package):
  Could not find a package configuration file provided by "Microsoft.GSL"
  with any of the following names:

    Microsoft.GSLConfig.cmake
    microsoft.gsl-config.cmake

  Add the installation prefix of "Microsoft.GSL" to CMAKE_PREFIX_PATH or set
  "Microsoft.GSL_DIR" to a directory containing one of the above files.  If
  "Microsoft.GSL" provides a separate development package or SDK, be sure it
  has been installed.

**Comment by jimcavoy** [comment]
As stated in the README.md, to generate the build environment, enter the following:
```
cmake --preset=linux-base
```

**Comment by Harrold** [comment]
Running cmake --preset=linux-base prings the following

cmake --preset=linux-base
Preset CMake variables:

  CMAKE_BUILD_TYPE="Debug"
  CMAKE_CXX_COMPILER="/usr/bin/g++"
  CMAKE_C_COMPILER="/usr/bin/gcc"
  CMAKE_EXPORT_COMPILE_COMMANDS="TRUE"
  CMAKE_TOOLCHAIN_FILE="/scripts/buildsystems/vcpkg.cmake"

CMake Error at /usr/share/cmake/Modules/CMakeDetermineSystem.cmake:152 (message):
  Could not find toolchain file: /scripts/buildsystems/vcpkg.cmake
Call Stack (most recent call first):
  CMakeLists.txt:6 (project)


CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
-- Configuring incomplete, errors occurred!

This is the same message on every project.

**Comment by jimcavoy** [comment]
On the terminal type the following to see __make__ is installed on your system:
```
which make
```

**Comment by Harrold** [comment]
It is.

which make
/bin/make


make
make: *** No targets specified and no makefile found.  Stop.

**Comment by jimcavoy** [comment]
The Prerequisites section in the README.md file mentions that the vcpkg system must be installed and configured.

**Comment by Harrold** [comment]
Is vcpkg available for Linux?

**Comment by jimcavoy** [comment]
Yes, vcpkg is available on Linux.  ConfBind needs vcpkg to download depended components such as boost libraries.

**Comment by Harrold** [comment]
Got it, thank you. Have to find out following issue /bin/ld: /usr/local/lib/libexip.a(EXIParser.c.o): relocation R_X86_64_PC32 against symbol `errorCodeStrings' can not be used when making a shared object; recompile with -fPIC

**Comment by jimcavoy** [comment]
Rebuild exip project with the CMake file from the ConfLabelReader, https://github.com/jimcavoy/ConfLabelReader/tree/main/exipCMake.  It will add the -fPIC compiler flag.

**Comment by Harrold** [comment]
Ah! Done! Works. I have ConfBind, ConfLabelReader, IReflex and Piper built at this point.

**Comment by jimcavoy** [comment]
Great.  ConfLabelReader will not work with ConfBind yet.  It assumes the label is EXI encoded.  I am updating ConfLabelReader to work with XML text-formated labels.  Here is the https://github.com/jimcavoy/ConfLabelReader/tree/12-read-xml-labels I am working on to update its behavior.  It should be done in the New Year.

Next, I will work on porting ClipPub to Linux.  This is the last application.

**Comment by Harrold** [comment]
That link somehow gives me 404 not found. Github is weird.

**Comment by jimcavoy** [comment]
Yes, merged this branch into main and deleted it.   Do a fetch and pull on main to get the latest.  ConfLabelReader can now read XML labels from a transport stream.

**Comment by jimcavoy** [comment]
Close issue.  To build klvp, you need vcpkg.
